src/index.js

```diff
@@ -154,19 +155,41 @@ function renderPdf(tag) {
   }
 
   const result = {};
+  const isTopLevel = true;
 
   children.forEach(child => {
     const resolvedChild = resolve(child, context);
     result[resolvedChild.elementName] = resolveChildren(
       resolvedChild,
       context,
-      true,
+      isTopLevel,
     );
   });
+  let { header, footer } = result;
+
+  if (has(attributes, 'header') && typeof attributes.header === 'function') {
+    header = (currentPage, pageCount, pageSize) =>
+      resolveChildren(
+        attributes.header(currentPage, pageCount, pageSize),
+        context,
+        isTopLevel,
+      );
+  }
+
+  if (has(attributes, 'footer') && typeof attributes.footer === 'function') {
+    footer = (currentPage, pageCount) =>
+      resolveChildren(
+        attributes.footer(currentPage, pageCount),
+        context,
+        isTopLevel,
+      );
+  }
 
   return {
     ...result,
     ...attributes,
+    header,
+    footer,
```

Comment:
I’m on mobile so I could be misreading this, but wouldn’t it clear the header and footer properties on the return value if they are set but _aren’t_ functions?

Reply:
Maybe worth adding some tests for this alternative case to be safe :)

Reply:
@mtlewis This won't override the existing footer & header if they exist but no function is provided since we initialize from the result above.

Will add a test to verify

Reply:
Ah yes, I see that now :)